testing/test_plugins/InteractionTestAsync.py

```diff
@@ -0,0 +1,197 @@
+            [self.pocket_atom.index],
+            [self.ligand_atom.index])
+        await interaction1.upload()
+        interactions = await Interaction.get()
+        assert len(interactions) == 1
+        interaction1.destroy()
+
+        interactions = await Interaction.get()
+        assert len(interactions) == 0
+        Logs.message("InteractionTest: test_upload passed")
+
+    async def test_toggle_visibility(self):
+        interactions = await Interaction.get()
+        assert len(interactions) == 0
+        interaction = Interaction(
+            enums.InteractionKind.HydrogenBond,
+            [self.pocket_atom.index],
+            [self.ligand_atom.index])
+        assert interaction.visible == True
+        await interaction.upload()
+
+        # Check that interaction is visible
+        interactions = await Interaction.get()
+        assert len(interactions) == 1
+        interaction = interactions[0]
+        assert interaction.visible is True  # This fails.
+
+        interaction.visible = False
+        await interaction.upload()
+        interactions = await Interaction.get()
+        assert len(interactions) == 1
+        interaction = interactions[0]
+        assert interaction.visible is False  # This fails
+        Interaction.destroy_multiple([interaction])
+
+    async def test_filter_by_kind(self):
+        interactions = await Interaction.get()
+        assert len(interactions) == 0
+        interaction1 = Interaction(
+            enums.InteractionKind.HydrogenBond,
+            [self.pocket_atom.index],
+            [self.ligand_atom.index])
+        interaction2 = Interaction(
+            enums.InteractionKind.Covalent,
+            [self.pocket_atom.index],
+            [self.ligand_atom.index])
+
+        # Draw another interaction connecting a different atom
+        ligand_atom2 = next(atom for atom in self.ligand_res.atoms if atom.index != self.ligand_atom.index)
+        interaction3 = Interaction(
+            enums.InteractionKind.Covalent,
+            [self.pocket_atom.index],
+            [ligand_atom2.index])
+        await Interaction.upload_multiple([interaction1, interaction2, interaction3])
+
+        interactions = await Interaction.get()
+        assert len(interactions) == 3
+
+        covalent_interactions = await Interaction.get(type_filter=enums.InteractionKind.Covalent)
+        assert len(covalent_interactions) == 2
+
+        Interaction.destroy_multiple([interaction1, interaction2, interaction3])
+        interactions = await Interaction.get()
+        assert len(interactions) == 0
+        Logs.message("InteractionTest: test_upload_multiple passed")
+
+    async def setup_test_workspace(self):
+        """Create fresh workspace and load it into Nanome."""
+        self.pdb_file = os.path.join(test_pdbs, '1tyl.pdb')
+        workspace = Workspace()
+        self.update_workspace(workspace)
+        await asyncio.sleep(1)  # Wait for workspace to be loaded. V annoying.
```

Comment:
Since `update_workspace` has a callback now, can this be replaced with `await self.update_workspace(workspace)`?